### ISSUE TITLE
Can we manage the model separately?

### ISSUE THREAD

**Issue by jikkuatwork** [VishnuPJ/Malayalam_Speech2Speech_Vision_Text_Chatbot#1]
Instead of relying on `HF`, can we link to the `gguf` file? I tried converting the code using claude, but its failing. Any chance you can give some pointers to use a local file than going via HF?

**Comment by VishnuPJ** [comment]
If you want to use a gguf file for the language model just start the llama cpp server and uncomment this function https://github.com/VishnuPJ/Malayalam_Speech2Speech_Vision_Text_Chatbot/blob/main/Malayalam_chatbot.py#L420

**Comment by jikkuatwork** [comment]
No, not just language model all other models. Now, since download is managed by HF. I was wondering whether we can download the respective models and point to the files?

Also, can't the language model be configured to use a hosted model: GPT4/Claude/ollama? Perhaps using ollama & adhering to openai API that will significantly simplify the code while enabling to swap any model.

**Comment by VishnuPJ** [comment]
> No, not just language model all other models. Now, since download is managed by HF. I was wondering whether we can download the respective models and point to the files?
> 

Yes , you can download the whole model from HF and just give the path to that folder instead of HF repo.
Use something like this to download the model,
```
from huggingface_hub import snapshot_download
model_id = "meta-llama/Llama-2-7b-hf"
snapshot_download(repo_id = model_id , local_dir = r"/llama_7b_hf",local_dir_use_symlinks = False,revision = "main")
```

> Also, can't the language model be configured to use a hosted model: GPT4/Claude/ollama? Perhaps using ollama & adhering to openai API that will significantly simplify the code while enabling to swap any model.

Right now I'm deploying the language and vision models and calling the API. You can use any models you want; just change the HF repo name.
https://github.com/VishnuPJ/Malayalam_Speech2Speech_Vision_Text_Chatbot/blob/main/flask_server.py#L26

**Comment by jikkuatwork** [comment]
Thanks, let me see
